File: 단계별로 풀어보기/1차원 배열/10818/최소, 최대.py
import sys

def findMaxMin(inputList):
    MAX=inputList[0]
    MIN=inputList[0]
    for i in range(1,len(inputList)):
        if MAX < inputList[i]:
            MAX = inputList[i]
        if MIN > inputList[i]:
            MIN = inputList[i]

    print("%d %d" %(MIN, MAX))

def init():
    N=int(sys.stdin.readline())
    inputList=list(map(int,sys.stdin.readline().split()))
    findMaxMin(inputList)

init()

File: 단계별로 풀어보기/1차원 배열/10818/최소, 최대.cpp
#include <iostream>
#include <list>
using namespace std;

list<int> makeInputList() {
	int N, a;
	list<int> inputList;
	cin >> N;
	for(int i=0; i < N; i++){
		cin >> a;
		inputList.push_back(a);
	}

	return inputList;
}

int findMax(list<int> inputList){
	int max;
	max = inputList.front();
	list<int>::iterator it;
	for (it = inputList.begin(); it != inputList.end(); it++) {
		if (max < *it)
			max = *it;
	}

	return max;
}

int findMin(list<int> inputList) {
	int min;
	min = inputList.front();
	list<int>::iterator it;
	for (it = inputList.begin(); it != inputList.end(); it++) {
		if (min > *it)
			min = *it;
	}

	return min;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	list<int> list = makeInputList();
	cout << findMin(list) <<' '<< findMax(list);
}
